src/main/java/com/checkmarx/sdk/dto/ScanResults.java

```diff
@@ -256,7 +258,12 @@ public boolean equals(Object o) {
             XIssue issue = (XIssue) o;
 
             if (!vulnerability.equals(issue.vulnerability)) return false;
-            return filename.equals(issue.filename);
+            if (groupBySeverity) {
+        	if (!filename.equals(issue.filename)) return false;
+        	return severity.equals(issue.severity);
```

Comment:
```suggestion
        	return filename.equals(issue.filename) && severity.equals(issue.severity);
```

Reply:
Yes, this looks better. Will make this change.